### PR TITLE
Added event for media unavailable.

### PR DESCRIPTION
**Pull request by RocheL**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/VIH-4416
### Change description ###
Added an alert for an answer 'no' for camera, microphone and video screens.
**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: VideoAPI/VideoApi.Domain/Enums/EventType.cs [hunk 1]
```diff
@@ -12,6 +12,7 @@ public enum EventType
         Leave,
         Consultation,
         JudgeAvailable,
-        MediaPermissionDenied
+        MediaPermissionDenied,
+        MediaProblem
```
Comment (shaed-parkar) [inline VideoAPI/VideoApi.Domain/Enums/EventType.cs]: Change to 'SelfTestFailed' event type

File: VideoAPI/VideoApi.Events/Handlers/MediaProblemEventhandler.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+using VideoApi.Domain.Enums;
+using VideoApi.Events.Handlers.Core;
+using VideoApi.Events.Hub;
+using VideoApi.Events.Models;
+using VideoApi.Events.ServiceBus;
+using TaskStatus = VideoApi.Domain.Enums.TaskStatus;
+
+namespace VideoApi.Events.Handlers
+{
+    public class MediaProblemEventHandler : EventHandlerBase
```
Comment (shaed-parkar) [inline VideoAPI/VideoApi.Events/Handlers/MediaProblemEventhandler.cs]: Rename to the event type name discussed above

File: VideoAPI/VideoApi.Events/Handlers/MediaProblemEventhandler.cs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        {
+        }
+
+        public override EventType EventType => EventType.MediaProblem;
+
+        protected override async Task PublishStatusAsync(CallbackEvent callbackEvent)
+        {
+            var query = new GetTasksForConferenceQuery(SourceConference.Id);
+            var tasks = await QueryHandler.Handle<GetTasksForConferenceQuery, List<Domain.Task>>(query);
+            var task = tasks.SingleOrDefault(x => x.Type == TaskType.Participant
```
Comment (shaed-parkar) [inline VideoAPI/VideoApi.Events/Handlers/MediaProblemEventhandler.cs]: This logic will not work, if a participant already has an open 'ToDo' task this task this not be added. You need to verify against the body as well.

File: VideoAPI/VideoApi.Events/Handlers/MediaProblemEventhandler.cs [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        protected override async Task PublishStatusAsync(CallbackEvent callbackEvent)
+        {
+            var query = new GetTasksForConferenceQuery(SourceConference.Id);
+            var tasks = await QueryHandler.Handle<GetTasksForConferenceQuery, List<Domain.Task>>(query);
+            var task = tasks.SingleOrDefault(x => x.Type == TaskType.Participant 
+                && x.OriginId == SourceParticipant.Id && x.Status != TaskStatus.ToDo);
+            if (task == null)
+            {
+                var command = new AddTaskCommand(SourceConference.Id, SourceParticipant.Id, 
+                    "Media problem", TaskType.Participant);
```
Comment (shaed-parkar) [inline VideoAPI/VideoApi.Events/Handlers/MediaProblemEventhandler.cs]: this should have the reason for failure

File: VideoAPI/VideoApi.Domain/Enums/EventType.cs (hunk 1 above)
Comment (RocheL) [reply VideoAPI/VideoApi.Domain/Enums/EventType.cs]: Updated EventType

File: VideoAPI/VideoApi.Events/Handlers/MediaProblemEventhandler.cs (hunk 2 above)
Comment (RocheL) [reply VideoAPI/VideoApi.Events/Handlers/MediaProblemEventhandler.cs]: Updated the handler name

File: VideoAPI/VideoApi.Events/Handlers/MediaProblemEventhandler.cs (hunk 4 above)
Comment (RocheL) [reply VideoAPI/VideoApi.Events/Handlers/MediaProblemEventhandler.cs]: updated to take the string from reason property

File: VideoAPI/VideoApi.Events/Handlers/MediaProblemEventhandler.cs (hunk 3 above)
Comment (RocheL) [reply VideoAPI/VideoApi.Events/Handlers/MediaProblemEventhandler.cs]: updated to check the body field information

File: VideoAPI/VideoApi.Events/Handlers/SelfTestFailedEventhandler.cs [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+        }
+
+        public override EventType EventType => EventType.SelfTestFailed;
+
+        protected override async Task PublishStatusAsync(CallbackEvent callbackEvent)
+        {
+            var query = new GetTasksForConferenceQuery(SourceConference.Id);
+            var tasks = await QueryHandler.Handle<GetTasksForConferenceQuery, List<Domain.Task>>(query);
+            var task = tasks.SingleOrDefault(x => x.Type == TaskType.Participant 
+                && x.OriginId == SourceParticipant.Id && x.Body == callbackEvent.Reason);
```
Comment (shaed-parkar) [inline VideoAPI/VideoApi.Events/Handlers/SelfTestFailedEventhandler.cs]: include tasks that are ToDo too


`tasks.SingleOrDefault(x => x.Type == TaskType.Participant && x.Status == TaskStatus.ToDo
                && x.OriginId == SourceParticipant.Id && x.Body == callbackEvent.Reason);`

Comment (RocheL) [reply VideoAPI/VideoApi.Events/Handlers/SelfTestFailedEventhandler.cs]: updated to include the todo tasks as well.

Comment (shaed-parkar) [reply VideoAPI/VideoApi.Events/Handlers/SelfTestFailedEventhandler.cs]: you seem to be missing a scenario to ensure the AddTaskCommand is not called when an existing pending task is found.

Comment (RocheL) [reply VideoAPI/VideoApi.Events/Handlers/SelfTestFailedEventhandler.cs]: added the test case to not call addTaskCommand if a task exists

**Pull request merged**